src/mono/mono/metadata/class.c

```diff
@@ -5855,6 +5855,26 @@ is_valid_family_access (MonoClass *access_klass, MonoClass *member_klass, MonoCl
 	return TRUE;
 }
 
+static gboolean
+ignores_access_checks_to (MonoAssembly *accessing, MonoAssembly *accessed)
+{
+	if (accessing == accessed)
+		return TRUE;
```

Comment:
Hmm.  I dont' think this is right.  I think this will make all `private` members accessible.

Reply:
Wouldn't `can_access_internals` be broken too then?

Reply:
No because `can_access_internals` is only called for `internal` or `protected` or `private protected`.  IgnoresAccessChecks is actually more powerful (c.f. the re-enabled tests) it will let you reach in to get `private` methods too.

Reply:
I would be somewhat surprised if that did not trigger a test failure, but maybe it's disabled for other reasons. If a test like that does not exist, it also might be worth explicitly adding one.

Reply:
Working on adding one.  I coudln't find aything like it.  There's not a lot of mentions of MethodAccessException in the codebase!

Reply:
Added 2 more tests.  Now we check that calling `internal`/`private` methods across assemblies throws a `MethodAccessException` and that calling a `private` method in the same assembly also throws.